### DISCUSSION TITLE
Unable to have Watermask shown

### DISCUSSION THREAD

**Discussion by msavazzi** [MahApps/MahApps.Metro#4017] [Q&A]
Team, sorry to bother but I have a very simple usercontrol but Watermark does never show:

<Page
    x:Class="TitoDoc2020.Views.Test1"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:local="clr-namespace:TitoDoc2020.Views"
    xmlns:mah="http://metro.mahapps.com/winfx/xaml/controls"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    Title="Test1"
    d:DesignHeight="450"
    d:DesignWidth="800"
    mc:Ignorable="d">

    <Page.Resources>
        <Thickness x:Key="ControlMargin">0 5 0 0</Thickness>
    </Page.Resources>
    
    <ScrollViewer HorizontalScrollBarVisibility="Hidden" VerticalScrollBarVisibility="Auto">
        <StackPanel>
            <TextBox
                        Margin="{StaticResource ControlMargin}"
                        mah:TextBoxHelper.Watermark="Watermark"
                        IsEnabled="False" />

        </StackPanel>
    </ScrollViewer>
</Page>

As I've created the project with the Template everything should be there:

<Application
    x:Class="TitoDoc2020.App"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:d1p1="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:mah="http://metro.mahapps.com/winfx/xaml/controls"
    xmlns:vm="clr-namespace:TitoDoc2020.ViewModels"
    d1p1:Ignorable="d"
    DispatcherUnhandledException="OnDispatcherUnhandledException"
    Exit="OnExit"
    Startup="OnStartup">

    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary Source="/Styles/_FontSizes.xaml" />
                <ResourceDictionary Source="/Styles/_Thickness.xaml" />
                <ResourceDictionary Source="/Styles/MetroWindow.xaml" />
                <ResourceDictionary Source="/Styles/TextBlock.xaml" />
                <ResourceDictionary Source="/Styles/TitoDocStyles.xaml" />
                
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Controls.xaml" />
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Fonts.xaml" />
                
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Themes/Dark.Blue.xaml" />
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Themes/Light.Blue.xaml" />
            </ResourceDictionary.MergedDictionaries>
            <vm:ViewModelLocator x:Key="Locator" d:IsDataSource="True" />
            <ControlTemplate x:Key="DefaultErrorTemplate">
                <Grid x:Name="container">
                    <Grid.ColumnDefinitions>
                        <ColumnDefinition Width="*" />
                    </Grid.ColumnDefinitions>

                    <AdornedElementPlaceholder
                        Name="customAdorner"
                        Grid.Column="0"
                        VerticalAlignment="Center" />

                    <Border
                        x:Name="ValidationErrorElement"
                        BorderBrush="#FFDB000C"
                        BorderThickness="1"
                        CornerRadius="1"
                        ToolTip="{Binding ElementName=customAdorner, Path=AdornedElement.(Validation.Errors)[0].ErrorContent}">
                        <Grid
                            Width="12"
                            Height="12"
                            Margin="1,-4,-4,0"
                            HorizontalAlignment="Right"
                            VerticalAlignment="Top"
                            Background="Transparent">
                            <Path
                                Margin="1,3,0,0"
                                Data="M 1,0 L6,0 A 2,2 90 0 1 8,2 L8,7 z"
                                Fill="#FFDC000C" />
                            <Path
                                Margin="1,3,0,0"
                                Data="M 0,0 L2,0 L 8,6 L8,8"
                                Fill="#ffffff" />
                        </Grid>
                    </Border>

                    <Border
                        Grid.Column="1"
                        Width="15"
                        Height="15"
                        Margin="0,0,5,0"
                        HorizontalAlignment="Right"
                        VerticalAlignment="Center"
                        Background="Red"
                        CornerRadius="10"
                        ToolTip="{Binding ElementName=customAdorner, Path=AdornedElement.(Validation.Errors)[0].ErrorContent}">
                        <TextBlock
                            HorizontalAlignment="center"
                            VerticalAlignment="center"
                            FontWeight="Bold"
                            Foreground="white"
                            Text="!" />
                    </Border>
                </Grid>
            </ControlTemplate>
            <Style TargetType="{x:Type TextBox}">
                <Setter Property="Validation.ErrorTemplate" Value="{StaticResource DefaultErrorTemplate}" />
            </Style>
            <Style TargetType="{x:Type PasswordBox}">
                <Setter Property="Validation.ErrorTemplate" Value="{StaticResource DefaultErrorTemplate}" />
            </Style>
            <Style
                x:Key="DescriptionHeaderStyle"
                BasedOn="{StaticResource {x:Type Label}}"
                TargetType="{x:Type Label}">
                <Setter Property="FontSize" Value="22" />
                <Setter Property="HorizontalAlignment" Value="Center" />
            </Style>

            <Style BasedOn="{StaticResource MahApps.Styles.CustomValidationPopup}" TargetType="{x:Type mah:CustomValidationPopup}">
                <Setter Property="CloseOnMouseLeftButtonDown" Value="False" />
                <Setter Property="ShowValidationErrorOnMouseOver" Value="False" />
            </Style>

            
            <Storyboard x:Key="MahApps.Storyboard.Expander.Collapse.Fast">
                <DoubleAnimation
                    Storyboard.TargetName="ExpandSite"
                    Storyboard.TargetProperty="(UIElement.Opacity)"
                    From="1"
                    To="0"
                    Duration="0:0:0.1" />
                <ObjectAnimationUsingKeyFrames Storyboard.TargetName="ExpandSite" Storyboard.TargetProperty="(UIElement.Visibility)">
                    <DiscreteObjectKeyFrame KeyTime="0:0:0.1" Value="{x:Static Visibility.Collapsed}" />
                </ObjectAnimationUsingKeyFrames>
            </Storyboard>

            <Style BasedOn="{StaticResource MahApps.Styles.Expander}" TargetType="{x:Type Expander}">
                <Setter Property="mah:ExpanderHelper.CollapseStoryboard" Value="{StaticResource MahApps.Storyboard.Expander.Collapse.Fast}" />
            </Style>

            <Style x:Key="MahApps.Styles.MetroHeader.Horizontal" TargetType="mah:MetroHeader">
                <Setter Property="Template">
                    <Setter.Value>
                        <ControlTemplate TargetType="mah:MetroHeader">
                            <Grid>
                                <Grid.ColumnDefinitions>
                                    <ColumnDefinition Width="Auto" SharedSizeGroup="HeaderGroup" />
                                    <ColumnDefinition Width="*" />
                                </Grid.ColumnDefinitions>
                                <Grid Grid.Column="0" Background="{TemplateBinding mah:HeaderedControlHelper.HeaderBackground}">
                                    <mah:ContentControlEx
                                        x:Name="PART_Header"
                                        Margin="{TemplateBinding mah:HeaderedControlHelper.HeaderMargin}"
                                        HorizontalAlignment="{TemplateBinding mah:HeaderedControlHelper.HeaderHorizontalContentAlignment}"
                                        VerticalAlignment="{TemplateBinding mah:HeaderedControlHelper.HeaderVerticalContentAlignment}"
                                        Content="{TemplateBinding Header}"
                                        ContentCharacterCasing="{TemplateBinding mah:ControlsHelper.ContentCharacterCasing}"
                                        ContentStringFormat="{TemplateBinding HeaderStringFormat}"
                                        ContentTemplate="{TemplateBinding HeaderTemplate}"
                                        ContentTemplateSelector="{TemplateBinding HeaderTemplateSelector}"
                                        FontFamily="{TemplateBinding mah:HeaderedControlHelper.HeaderFontFamily}"
                                        FontSize="{TemplateBinding mah:HeaderedControlHelper.HeaderFontSize}"
                                        FontStretch="{TemplateBinding mah:HeaderedControlHelper.HeaderFontStretch}"
                                        FontWeight="{TemplateBinding mah:HeaderedControlHelper.HeaderFontWeight}"
                                        Foreground="{TemplateBinding mah:HeaderedControlHelper.HeaderForeground}"
                                        IsTabStop="False"
                                        RecognizesAccessKey="{TemplateBinding mah:ControlsHelper.RecognizesAccessKey}"
                                        SnapsToDevicePixels="{TemplateBinding SnapsToDevicePixels}" />
                                </Grid>
                                <Grid Grid.Column="1" Background="{TemplateBinding Background}">
                                    <ContentPresenter
                                        x:Name="PART_Content"
                                        Margin="{TemplateBinding Padding}"
                                        HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
                                        VerticalAlignment="{TemplateBinding VerticalContentAlignment}"
                                        ContentSource="Content"
                                        SnapsToDevicePixels="{TemplateBinding SnapsToDevicePixels}" />
                                </Grid>
                            </Grid>
                        </ControlTemplate>
                    </Setter.Value>
                </Setter>
                <Setter Property="mah:HeaderedControlHelper.HeaderMargin" Value="0 0 4 0" />
                <Setter Property="mah:HeaderedControlHelper.HeaderVerticalContentAlignment" Value="Center" />
            </Style>

            <Style
                x:Key="MahApps.Styles.ColorPalette.ColorPickerDropDown"
                BasedOn="{StaticResource MahApps.Styles.ColorPalette}"
                TargetType="{x:Type mah:ColorPalette}">
                <Setter Property="mah:ControlsHelper.ContentCharacterCasing" Value="Normal" />
            </Style>

        </ResourceDictionary>
    </Application.Resources>
</Application>

I've even updated to latest version of mahapps libraries.

Thank you
M

**Comment by punker76** [comment]
@msavazzi  The problem is that you override the default `TextBox` style with your own. You forgot to inherit the styles:

```
        <Style TargetType="{x:Type TextBox}">
            <Setter Property="Validation.ErrorTemplate" Value="{StaticResource DefaultErrorTemplate}" />
        </Style>
        <Style TargetType="{x:Type PasswordBox}">
            <Setter Property="Validation.ErrorTemplate" Value="{StaticResource DefaultErrorTemplate}" />
        </Style>
```

which it should be:

```
        <Style TargetType="{x:Type TextBox}" BasedOn="{StaticResource {x:Type TextBox}}">
            <Setter Property="Validation.ErrorTemplate" Value="{StaticResource DefaultErrorTemplate}" />
        </Style>
        <Style TargetType="{x:Type PasswordBox}" BasedOn="{StaticResource {x:Type PasswordBox}}">
            <Setter Property="Validation.ErrorTemplate" Value="{StaticResource DefaultErrorTemplate}" />
        </Style>
```

**Reply by msavazzi** [reply]
Amazing thank you! I was declaring that as I'm using the MvvmValidation component that is great. This because I prefer not to use the IDataErrorInfo method that is implemented in MahApps. Maybe we can add the correct support in MahApps 👍
